intake/source/base.py

```diff
@@ -280,6 +280,26 @@ def hvplot(self):
         """
         return self.plot
 
+    def persist(self, name, **kwargs):
+        """Save data from this source to local persistent storage"""
+        from ..container import container_map
+        import datetime
+        method = container_map[self.container]._persist
+        out = method(self, name, **kwargs)
+        out.description = self.description
+        metadata = {'type': 'persisted_dataset',
+                    'timestamp': datetime.datetime.now().isoformat(),
+                    'previous_metadata': self.metadata,
```

Comment:
What would the consequences be of leaving `metadata = self.metadata` and putting the persistence-related info _beside_ it in a new attribute rather than nesting?